Added Japanease Localization

puziro opened the pull request:
To RFO-BASIC master maintainer;

I am not good at github operation. Please review my update.

puziro committed:
Japanese Loacalization

The Japanese localization of the user interface has been added.

puziro committed:
Japanese Loaclization 2

The Japanese localization of the user interface has been added.

puziro committed:
Japanese Localization 3

The Japanese localization of the user interface has been added.

puziro committed:
Localize and Eclipse to Android Studio

Localization and  imported BASIC! Eclipse project  into Android Studio import an Eclipse project into Android Studio V2.3.3

puziro committed:
Update README.txt

jMarcS commented:
Hello, Yamazaki さん,

I apologize for being very slow to reply to you.

I truly appreciate the work you have done to provide Japanese localization for the BASIC! program source. I would like to pull those changes into the main RFO-BASIC repository.
However, I am not yet prepared to change the repository for Android Studio.

Unfortunately, I am not very good with GitHub operation, either. I have not learned how to modify your pull request to use only the parts that I want. I want your first three commits, plus the four res files from your fourth commit. Please tell me if there are any other files that have changed for localization.

I can easily copy your changes into my workspace and simply commit them. That is the simplest way to get your changes into the mainstream. However, if I do that, the GitHub change log will not show that you did all of the work.

I will certainly give you full credit. I will put comments here in the GitHub discussion, in the the BASIC! forum, in the source code, in all checkout comments, and in the De Re BASIC! manual. But if you want the GitHub change log to show a completed Pull Request, we will have to figure out how to do that.

What do you want to do?

puziro commented:
Hello, Marc-san

Thank you for your reply.

In fact, this is my first Github pull request and I am completely new to Github operation. (Also I am new to Android Studio.) Therefore I might be wrong. 

So,  I support following your idea. I don't mind if the GitHub change log shows my work or not. I think it is better to release new version quickly.
>I can easily copy your changes into my workspace and simply commit them. That is the simplest way to >get your changes into the mainstream. However, if I do that, the GitHub change log will not show that >you did all of the work.

Today I push my local data in Android studio to Github https://github.com/puziro/Basic/tree/Basic!ML . This is my latest data. Would you please check my update ??  
I added following changes to your app as far as I can remember. 
- Added sample program folder and some sample programs for Japanese 
- Replace Basic! icons to Basic!ML icons ( ML stands for Multi Language )
- Add 'Basic! user group site in Japan' link to about.xml and about.java.
- Add Values-Ja folder and related files  
- Change Values/string.xml file for new string data
- Add Japanese msg for ja environment to Basic.java
- Replace some strings to getSrting() for Japanese   in  run.java , loadfile.java , search.java , editor.java
 ( I would like to replace all error msgs to getString() but I didn't yet. If my changes were appropriate, I would update them later.)

This is my Basic!ML apk file for test purpose. For your reference. 
[BasicML-ml-release.zip](https://github.com/RFO-BASIC/Basic/files/2467609/BasicML-ml-release.zip)
If you need Android studio 2.3.3 project file for Basic!ML , please let me know. 

I hope this helps you.

Regards, Hiro

jMarcS commented:
Thank you, Hiroさん. No, there is no need for another pull request. I won't change to Android Studio with this release, so I am manually merging your changes into Eclipse. I know we should have moved to AS a long time ago! It will have to wait one more release.
I am required to build this update with API 26 or later. I'm still trying to find a way to do that and preserve support for API 7, as BASIC! always has. I think Google has made it impossible.

puziro commented:
Hi, Mark-san,

Thank you. I am looking forward new version Basic! with Japanese.  
Please let me know if there are incomplete Japanese translation. I will correct them.

 >so I am manually merging your changes into Eclipse. 
--->I am sorry that I exported your project to AS. ( I never used Eclipse. )

>I am required to build this update with API 26 or later. I'm still trying to find a way to do that and preserve support for API 7, as BASIC! always has. I think Google has made it impossible.
----> They take a change for the worse even if there are reasons for them. They are responsible for backward compatibility.

Regards, Hiro
